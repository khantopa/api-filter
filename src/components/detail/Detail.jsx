import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './Detail.scss';

export default class Detail extends Component {
  constructor(props) {
    super(props)
    this.state = {
      contact: {},
      company: {}
    }
  }

  componentDidMount = () => {
    const id = this.props.match.params.id;
    axios.get(`http://165.22.60.22/api/v1/contacts/${id}`)
    .then(response => {
      this.setState({ contact: response.data.data, company: response.data.data.company })
    })
    .catch( err => console.log(err))
  }


  render() {
    const { name, email } = this.state.contact;
    const company = this.state.company;
    return (
      <div className="contact">
        <Link to='/contacts' className="button">Back</Link>
        <div className="contact_card">
          <h1>{ name }</h1>
          <h6>E-mail: { email }</h6>
          <div className="company">
            <p><span>Company ID:</span>&nbsp;&nbsp; { company.id }</p>
            <p><span>Company:</span>&nbsp;&nbsp; { company.name }</p>
            <p><span>Revenue:</span>&nbsp;&nbsp; { company.revenue }</p>
            <p><span>Country:</span>&nbsp;&nbsp; { company.country }</p>
          </div>
        </div>
      </div>
    )
  }
}
