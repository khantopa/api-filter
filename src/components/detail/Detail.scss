@import '../../styles/main.scss';

.contact {
  position: relative;
  height: 100vh;
  width: 100%;
  padding: 40px;
  box-sizing: border-box;
  background: $primary;
  .button {
    width: 100px;
    color: $light;
    background: $dark;
  }
  &_card {
    @include absPosition(50%, 50%);
    transform: translate(-50%, -50%);
    @include card(80vh, auto);
    background: rgba(255, 255, 255, .9);
    color: $primary;
    h1 {
      text-transform: uppercase;
      line-height: 38px;
      font-size: 30px;
      letter-spacing: .2em;
    }
    h6 {
      font-size: 14px;
      color: $dark;
    }
    .company {
      margin: 30px auto;
      text-align: left;
      p {
        font-size: 18px;
        color: $dark;
        span {
          font-weight: bold;
          color: $primary;
        }
      }
    }
  }
}

@media (max-width: 700px) {
  .contact {
    &_card {
      position: unset;
      transform: unset;
      @include card(100%, auto);
    }
  }
}