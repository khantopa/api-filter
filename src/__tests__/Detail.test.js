import Detail from '../components/detail/Detail';


let wrapper;
let params = { params: {id: 1}};

beforeEach(() => {
  wrapper = shallow(<Detail match={ params } /> ); 
});



it('should render Detail', () => {
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});




