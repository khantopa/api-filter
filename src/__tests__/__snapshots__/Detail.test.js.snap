// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Detail 1`] = `
<div
  className="contact"
>
  <Link
    className="button"
    to="/contacts"
  >
    Back
  </Link>
  <div
    className="contact_card"
  >
    <h1 />
    <h6>
      E-mail: 
    </h6>
    <div
      className="company"
    >
      <p>
        <span>
          Company ID:
        </span>
           
      </p>
      <p>
        <span>
          Company:
        </span>
           
      </p>
      <p>
        <span>
          Revenue:
        </span>
           
      </p>
      <p>
        <span>
          Country:
        </span>
           
      </p>
    </div>
  </div>
</div>
`;
